presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -437,14 +442,9 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
   // TODO(venkatra): Populate these memory stats as well.
   prestoTaskStats.revocableMemoryReservationInBytes = {};
 
-  // Set the lastTaskStatsUpdateMs to execution start time if it is 0.
-  if (lastTaskStatsUpdateMs == 0) {
-    lastTaskStatsUpdateMs = taskStats.executionStartTimeMs;
-  }
-
   const int64_t currentBytes = stats.currentBytes;
 
-  int64_t averageMemoryForLastPeriod =
+  const int64_t averageMemoryForLastPeriod =
       (currentBytes + lastMemoryReservation) / 2;
 
   prestoTaskStats.cumulativeUserMemory +=
```

Comment:
Can we move 
```
const double sinceLastPeriodMs = currentTimeMs - lastTaskStatsUpdateMs;
```
close to the code that actually uses "sinceLastPeriodMs" (here)?
Otherwise it is detached and not clear why we need it, hence I was asking "why double". 🙂